src/accountsdb/db.zig

```diff
@@ -2377,21 +2441,23 @@ pub const AccountsDB = struct {
         const file_map, var file_map_lg = self.file_map.readWithLock();
         defer file_map_lg.unlock();
 
-        const latest_full_snapshot_info, var latest_full_snapshot_info_lg = self.latest_full_snapshot_info.writeWithLock();
-        defer latest_full_snapshot_info_lg.unlock();
+        // lock this now such that, if under any circumstance this method was invoked twice in parallel
+        // on separate threads, there wouldn't be any overlapping work being done.
+        const p_maybe_latest_snapshot_info: *?SnapshotGenerationInfo, var latest_snapshot_info_lg = self.latest_snapshot_info.writeWithLock();
+        defer latest_snapshot_info_lg.unlock();
 
         std.debug.assert(zstd_buffer.len != 0);
         std.debug.assert(params.target_slot <= self.largest_flushed_slot.load(.monotonic));
 
         const full_hash, const full_capitalization = compute: {
-            check_cache: {
-                const p_maybe_first_full, var first_full_lg = self.account_hashes_and_lamports_cache.first_full.readWithLock();
-                defer first_full_lg.unlock();
+            check_first: {
+                const p_maybe_first, var first_lg = self.first_snapshot_load_info.readWithLock();
```

Comment:
should fix these names too -- p_maybe_first doesnt really mean anything

Reply:
dcab9d8f726aaa401cd18b4080e729704b11d3e7